cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/SingleThreadDatasetCache.java

```diff
@@ -268,6 +269,13 @@ private void discardSafely(Object dataset) {
         }
       }
     }
+
+    // CDAP-14326 check if at least one of the entries of the given dataset was invalidated from datasetCache. This
+    // means dataset provided was accquired through this context. Hence return without logging any warning.
+    if (isInvalidated) {
```

Comment:
You don't need this variable. Just check `if (accessTypes.size() < AccessType.values().length)`, as that means at least some access types have been removed.